### PR TITLE
Convert all files to ts

### PR DESCRIPTION
**Pull request by ninaandal**

### CODE REVIEW

File: types/index.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export interface AuthorProps {
+  name: string
+  picture: any
```
Comment (ninaandal) [inline types/index.tsx:3]: Not quite sure what the correct type here is

Comment (RitaDias) [reply types/index.tsx:3]: I would think it's time `image` perhaps? Or was that changed? 🤔

Comment (stipsan) [reply types/index.tsx:3]: This is technically a:
```
import {Image} from '@sanity/types'
```
But let's deal with this in a follow-up as it's not clear how to strike the best balance between:
- A: good detailed types that make it a lot of work to change your groq queries.
- B: friendly types that let you easily change groq queries, but TS can't know if the data groq returns matches what the components expect.

**Review by stipsan — APPROVED** [review]
Awesome work @ninaandal! ✨ 🎉

**Comment by kmelve** [comment]
#### 📝 Changed routes:


| [**/**](https://vercel-next-js-examples-cms-sanity-studio-v3-r8io2j3f4.sanity.build/) | [**/posts/\[slug\]**](https://vercel-next-js-examples-cms-sanity-studio-v3-r8io2j3f4.sanity.build/posts/[slug]) |
|:-:|:-:|
|<a href="https://vercel-next-js-examples-cms-sanity-studio-v3-r8io2j3f4.sanity.build/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/3T3qwS3-qdQTonyKao1DeWXt9CXTSR-BEkoEUpa1V7_qzICDfCRF2oLBngyD-SzTh4Ah8XReKTu8G_8ikukxMg.png">(view full size)</a> |<a href="https://vercel-next-js-examples-cms-sanity-studio-v3-r8io2j3f4.sanity.build/posts/[slug]">Screenshot of /posts/[slug]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/bB48kOGb885XJmez-et-3iCHkKg3yVZHr1PfatsTB-MVuskV0KOg2xNh844eAYgcuipI4E0bAJOW0cioaLT__A.png">(view full size)</a> |



And 3 other routes:
- [**/api/exit-preview**](https://vercel-next-js-examples-cms-sanity-studio-v3-r8io2j3f4.sanity.build/api/exit-preview)
- [**/api/preview**](https://vercel-next-js-examples-cms-sanity-studio-v3-r8io2j3f4.sanity.build/api/preview)
- [**/api/revalidate**](https://vercel-next-js-examples-cms-sanity-studio-v3-r8io2j3f4.sanity.build/api/revalidate)



Commit 361479b9fbc2e9e06f0c47ad3d0932e684cbc2f7 (https://vercel-next-js-examples-cms-sanity-studio-v3-r8io2j3f4.sanity.build).

**Pull request merged**